src/symbolize/gimli/parse_running_mmaps_unix.rs

```diff
@@ -81,46 +90,73 @@ impl FromStr for MapsEntry {
     // e.g.: "ffffffffff600000-ffffffffff601000 --xp 00000000 00:00 0                  [vsyscall]"
     // e.g.: "7f5985f46000-7f5985f48000 rw-p 00039000 103:06 76021795                  /usr/lib/x86_64-linux-gnu/ld-linux-x86-64.so.2"
     // e.g.: "35b1a21000-35b1a22000 rw-p 00000000 00:00 0"
-    //
-    // Note that paths may contain spaces, so we can't use `str::split` for parsing (until
-    // Split::remainder is stabilized #77998).
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-       fn error(msg: &str) -> &str {
+        // While there are nicer standard library APIs available for this, we aim for minimal code size.
+
+        let mut state = s;
+
+        fn parse_start<'a>(state: &mut &'a str) -> &'a str {
+            // Unsafe is unfortunately necessary to get the bounds check removed (for code size).
+
+            let start_idx = state.bytes().position(|b| b != b' ');
+            if let Some(start_idx) = start_idx {
+                // SAFETY: It comes from position, so it's in bounds.
+                //         It must be on a UTF-8 boundary as it's the first byte that isn't ' '.
+                *state = unsafe { state.get_unchecked(start_idx..) };
+            }
+            let match_idx = state.bytes().position(|b| b == b' ');
+            match match_idx {
+                None => {
+                    let result = *state;
+                    *state = "";
+                    result
+                }
+                Some(match_idx) => {
+                    // SAFETY: match_index comes from .bytes().position() of an ASCII character,
+                    //         so it's both in bounds and a UTF-8 boundary
+                    let result = unsafe { state.get_unchecked(..match_idx) };
+                    // SAFETY: Since match_idx is the ' ', there must be at least the end after it.
+                    *state = unsafe { state.get_unchecked((match_idx + 1)..) };
+                    result
+                }
+            }
+        }
```

Comment:
@hkBst This is an example of the kind of microoptimization that is both very hard to do in the compiler and also can reap significant benefits since here we care mostly about code size, even in the dead code, as this code size is essentially multiplied by *all* Rust binaries.

Reply:
@workingjubilee Thanks for thinking of me, although I'm not quite sure why you did.